src/typesystem.cpp

```diff
@@ -53,55 +53,57 @@ TypeSystemCore::TypeSystemCore() {
 		}
 
 	};
+
+	m_mono_core.unify_function = [&](int a, int b) {
+		a = m_mono_core.find_term(a);
+		b = m_mono_core.find_term(b);
+
+		if (a == b)
+			return;
+	
+		Unification::Core::TermData& a_data = m_mono_core.term_data[a];
+		Unification::Core::TermData& b_data = m_mono_core.term_data[b];
+
+		m_tf_core.unify(a_data.function_id, b_data.function_id);
+	};
 }
 
 
 void TypeSystemCore::print_type(MonoId mono, int d) {
```

Comment:
Maybe we should have a print_node in the unifier, instead?

Reply:
This is a good, yea